service/notification/NotificationProcessor.php

```diff
@@ -233,4 +263,34 @@ public function addMessage($notification)
 
         return true;
     }
+
+    /**
+     * Returns true if an Authorisation notification with success = true has already been processed before for the same
+     * merchant_reference
+     *
+     * @param $merchantReference
+     * @return bool
+     */
+    private function hasProcessedAuthorisationSuccessNotification($merchantReference)
+    {
+        $notificationModel = new AdyenNotification();
+
+        $processedNotifications = $notificationModel->getProcessedNotificationsByMerchantReference($merchantReference);
+
+        if (empty($processedNotifications)) {
+            return false;
+        }
+
+        $hasProcessedAuthorisedNotification = false;
+
+        foreach ($processedNotifications as $processedNotification) {
+            if (AdyenNotification::AUTHORISATION === $processedNotification['event_code'] &&
+                'true' === $processedNotification['success']
+            ) {
+                $hasProcessedAuthorisedNotification = true;
+            }
+        }
```

Comment:
we can add this as a method and when one item is found return it so we don't have to loop to all of the next notifications.